src/command/data.rs

```diff
@@ -646,64 +683,96 @@ fn key_info_byte(data: Option<KeyOrigin>) -> u8 {
-) -> Result<(), Status> {
-    // Key-Ref. : Sig = 1, Dec = 2, Aut = 3 (see §7.2.18)
-    ctx.reply.expand(&[
-        0x01,
-        key_info_byte(ctx.state.internal.key_origin(KeyType::Sign)),
-    ])?;
-    ctx.reply.expand(&[
-        0x02,
-        key_info_byte(ctx.state.internal.key_origin(KeyType::Dec)),
-    ])?;
-    ctx.reply.expand(&[
-        0x03,
-        key_info_byte(ctx.state.internal.key_origin(KeyType::Aut)),
-    ])?;
-    Ok(())
+fn key_info<const R: usize, T: trussed::Client>(mut ctx: Context<'_, R, T>) -> Result<(), Status> {
+    if let Ok(mut ctx) = ctx.load_state() {
+        // Key-Ref. : Sig = 1, Dec = 2, Aut = 3 (see §7.2.18)
+        ctx.reply.expand(&[
+            0x01,
+            key_info_byte(ctx.state.internal.key_origin(KeyType::Sign)),
+        ])?;
+        ctx.reply.expand(&[
+            0x02,
+            key_info_byte(ctx.state.internal.key_origin(KeyType::Dec)),
+        ])?;
+        ctx.reply.expand(&[
+            0x03,
+            key_info_byte(ctx.state.internal.key_origin(KeyType::Aut)),
+        ])?;
+        Ok(())
+    } else {
+        // If the state doesn't load, return placeholder so that gpg presents the option to factory reset
+        ctx.reply.expand(&hex!("010002000300"))
+    }
 }
 
 fn uif<const R: usize, T: trussed::Client>(
-    mut ctx: LoadedContext<'_, R, T>,
+    mut ctx: Context<'_, R, T>,
     key: KeyType,
 ) -> Result<(), Status> {
-    if !ctx.options.button_available {
-        warn!("GET DAT for uif without a button available");
-        return Err(Status::FunctionNotSupported);
-    }
+    if let Ok(mut ctx) = ctx.load_state() {
+        if !ctx.options.button_available {
+            warn!("GET DAT for uif without a button available");
+            return Err(Status::FunctionNotSupported);
+        }
 
-    let state_byte = ctx.state.internal.uif(key).as_byte();
-    let button_byte = general_feature_management_byte(ctx.options);
-    ctx.reply.expand(&[state_byte, button_byte])
+        let state_byte = ctx.state.internal.uif(key).as_byte();
+        let button_byte = general_feature_management_byte(ctx.options);
+        ctx.reply.expand(&[state_byte, button_byte])
+    } else {
+        // If the state doesn't load, return placeholder so that gpg presents the option to factory reset
+        ctx.reply.expand(&[
+            Uif::Disabled as u8,
+            general_feature_management_byte(ctx.options),
+        ])
+    }
 }
 
 fn cardholder_name<const R: usize, T: trussed::Client>(
-    mut ctx: LoadedContext<'_, R, T>,
+    mut ctx: Context<'_, R, T>,
 ) -> Result<(), Status> {
-    ctx.reply.expand(ctx.state.internal.cardholder_name())
+    if let Ok(mut ctx) = ctx.load_state() {
+        ctx.reply.expand(ctx.state.internal.cardholder_name())
+    } else {
+        // If the state doesn't load, return placeholder so that gpg presents the option to factory reset
+        ctx.reply
+            .expand(b"Card state corrupted. Factory reset recommended")
```

Comment:
I would remove the second sentence because I’d prefer people telling us in support if they run into this issue.  We could still document it in the release notes, but if this happens in a different context, I’d like to know.

Reply:
Should I recommend them to open an issue ? The limit is 39 characters so I don't think we can put a link.

Reply:
We could use the `URL` DO for that

Reply:
The limit at 39 characters doesn't seem to be an issue with GPG.